infra/cifuzz/run_fuzzers.py

```diff
@@ -27,61 +27,167 @@
+    if not self.config.fuzz_seconds or self.config.fuzz_seconds < 1:
+      logging.error(
+          'Fuzz_seconds argument must be greater than 1, but was: %s.',
+          self.config.fuzz_seconds)
+      return False
+
+    self.out_dir = os.path.join(self.config.workspace, 'out')
+    if not os.path.exists(self.out_dir):
+      logging.error('Out directory: %s does not exist.', self.out_dir)
+      return False
+
+    self.artifacts_dir = os.path.join(self.out_dir, 'artifacts')
+    if not os.path.exists(self.artifacts_dir):
+      os.mkdir(self.artifacts_dir)
+    elif (not os.path.isdir(self.artifacts_dir) or
+          os.listdir(self.artifacts_dir)):
+      logging.error('Artifacts path: %s exists and is not an empty directory.',
+                    self.artifacts_dir)
+      return False
+
+    self.fuzz_target_paths = utils.get_fuzz_targets(self.out_dir)
+    logging.info('Fuzz targets: %s', self.fuzz_target_paths)
+    if not self.fuzz_target_paths:
+      logging.error('No fuzz targets were found in out directory: %s.',
+                    self.out_dir)
+      return False
+
+    return True
+
+  def run_fuzz_target(self, fuzz_target_obj):  # pylint: disable=no-self-use
+    """Fuzzes with |fuzz_target_obj| and returns the result."""
+    # TODO(metzman): Make children implement this so that the batch runner can
+    # do things differently.
+    return fuzz_target_obj.fuzz()
+
+  @property
+  def quit_on_bug_found(self):
+    """Property that is checked to determine if fuzzing should quit after first
+    bug is found."""
+    raise NotImplementedError('Child class must implement method')
+
+  def get_fuzz_target_artifact(self, target, artifact_name):
+    """Returns the path of a fuzzing |artifact| named |artifact_name| for
+    |target|."""
+    artifact_name = target.target_name + '-' + artifact_name
+    return os.path.join(self.artifacts_dir, artifact_name)
+
+  def create_fuzz_target_obj(self, target_path, run_seconds):
+    """Returns a fuzz target object."""
+    return fuzz_target.FuzzTarget(target_path,
+                                  run_seconds,
+                                  self.out_dir,
+                                  self.config.project_name,
+                                  sanitizer=self.config.sanitizer)
+
+  def run_fuzz_targets(self):
+    """Runs fuzz targets. Returns True if a bug was found."""
+    fuzzers_left_to_run = len(self.fuzz_target_paths)
+
+    # Make a copy since we will mutate it.
+    fuzz_seconds = self.config.fuzz_seconds
+
+    min_seconds_per_fuzzer = fuzz_seconds // fuzzers_left_to_run
+    bug_found = False
+    for target_path in self.fuzz_target_paths:
+      run_seconds = max(fuzz_seconds // fuzzers_left_to_run,
+                        min_seconds_per_fuzzer)
+
+      target = self.create_fuzz_target_obj(target_path, run_seconds)
+      start_time = time.time()
+      testcase, stacktrace = self.run_fuzz_target(target)
+      fuzz_seconds -= time.time() - start_time
```

Comment:
Can this go negative, does it need min with zero.

Reply:
It's fine if it goes negative since we take a max. I explain why now in comments.